fix: repair chromium installation failure

CodeChenL opened the pull request:
remove dpkg-divert from older package before install

CodeChenL commented:
@RadxaYuntian request a review

RadxaYuntian commented on debian/rockchip-chromium-x11-utils.preinst:
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
maintainer script避免使用bash，因为从理论上debian可以使用其他的shell

RadxaYuntian commented on debian/rockchip-chromium-x11-utils.preinst:
```diff
@@ -0,0 +1,21 @@
+            if [[ "$(dpkg-divert --list $i)" == *"rockchip-chromium-x11-utils"* ]]
+            then
+                dpkg-divert --package rockchip-chromium-x11-utils --no-rename --remove "$i"
+                mv "$i.bak" "$i"
```
以前的postrm也没有这个mv，难道老版本不能正常卸载？https://github.com/radxa-pkg/rockchip-chromium-x11-utils/commit/d60092a8bdc469c48a98136f6bb3eb1111ced0ca

RadxaYuntian replied:
哦你用了`--no-rename`，为什么不像以前一样用`--rename`？

CodeChenL replied:
飞书上说了会报错，所以用了mv

RadxaYuntian replied:
--rename试过么

CodeChenL replied:
试过了

RadxaYuntian commented on debian/rockchip-chromium-x11-utils.preinst:
```diff
@@ -0,0 +1,21 @@
+
+        for i in $conflicts
+        do
+            if [[ "$(dpkg-divert --list $i)" == *"rockchip-chromium-x11-utils"* ]]
```
```suggestion
            if [ "$(dpkg-divert --listpackage "$i")" == "rockchip-chromium-x11-utils" ]
```
`[[]]`不是POSIX sh支持的语法，变量要双引号包住
`--listpackage`的话可以避免字符串匹配

RadxaYuntian commented:
也许可以用breaks自己旧版本而不是maintainer script来解决：https://www.debian.org/doc/debian-policy/ch-relationships.html#packages-which-break-other-packages-breaks

CodeChenL commented:
> 也许可以用breaks自己旧版本而不是maintainer script来解决：https://www.debian.org/doc/debian-policy/ch-relationships.html#packages-which-break-other-packages-breaks

我试一下

CodeChenL commented:
Breaks: rockchip-chromium-x11-utils (<< 0.2.0)可以升级成功，但是会有dpkg-divert数据库残留：
```bash
rock@rock-5b:~$ dpkg -S /usr/lib/chromium/chromium-wrapper
diversion by rockchip-chromium-x11-utils from: /usr/lib/chromium/chromium-wrapper
diversion by rockchip-chromium-x11-utils to: /usr/lib/chromium/chromium-wrapper.bak
chromium-x11: /usr/lib/chromium/chromium-wrapper
rock@rock-5b:~$
```

CodeChenL commented:
> Breaks: rockchip-chromium-x11-utils (<< 0.2.0)可以升级成功，但是会有dpkg-divert数据库残留：
```
 rock@rock-5b:~$ dpkg -S /usr/lib/chromium/chromium-wrapper diversion by rockchip-chromium-x11-utils from: /usr/lib/chromium/chromium-wrapper diversion by rockchip-chromium-x11-utils to: /usr/lib/chromium/chromium-wrapper.bak chromium-x11: /usr/lib/chromium/chromium-wrapper rock@rock-5b:~$
```

CodeChenL commented:
> > > Breaks: rockchip-chromium-x11-utils (<< 0.2.0)可以升级成功，但是会有dpkg-divert数据库残留：
> > 
> > 
> > ```
> >  rock@rock-5b:~$ dpkg -S /usr/lib/chromium/chromium-wrapper 
> > ```
> 
> diversion by rockchip-chromium-x11-utils from: /usr/lib/chromium/chromium-wrapper diversion by rockchip-chromium-x11-utils to: /usr/lib/chromium/chromium-wrapper.bak chromium-x11: /usr/lib/chromium/chromium-wrapper rock@rock-5b:~$
> 
> >

RadxaYuntian commented:
那在preinst/postinst里面把残留的东西给清理了，另外要注明是因为<<0.2.0的包添加的

CodeChenL commented:
> 那在preinst/postinst里面把残留的东西给清理了，另外要注明是因为<<0.2.0的包添加的

已修改

RadxaYuntian commented on debian/rockchip-chromium-x11-utils.lintian-overrides:
```diff
@@ -3,3 +3,6 @@ rockchip-chromium-x11-utils: improbable-bug-number-in-closes
 # Ignore troff warning
 rockchip-chromium-x11-utils: groff-message
+
+# Ignore lintian warning about package-relation-with-self(Breaks: rockchip-chromium-x11-utils (<< 0.2.0))
```
```suggestion
# We does conflict with our older versions
```

RadxaYuntian commented on debian/rockchip-chromium-x11-utils.preinst:
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+set -e
+
+# The chromium-wrapper is no longer provided in the current version, this script is used to clean up dpkg-divert for << version 0.2.0
```
这个移动到clean up之前。这个脚本以后可能会有别的功能，说明就不要放在开头
```suggestion
# `chromium-wrapper` is no longer provided
# Clean up dpkg-divert for version << 0.2.0
```

RadxaYuntian commented on debian/rockchip-chromium-x11-utils.preinst:
```diff
@@ -0,0 +1,16 @@
+# The chromium-wrapper is no longer provided in the current version, this script is used to clean up dpkg-divert for << version 0.2.0
+
+case $1 in
+    install|upgrade)
```
```suggestion
    upgrade)
```
只有升级才需要考虑清理遗留，另外检查一下旧版本是不是<<0.2.0
https://stackoverflow.com/a/18408532
https://manpages.debian.org/testing/dpkg-dev/deb-preinst.5.en.html#new-preinst~3

RadxaYuntian commented on debian/rockchip-chromium-x11-utils.preinst:
```diff
@@ -0,0 +1,16 @@
+case $1 in
+    install|upgrade)
+        conflicts="/usr/lib/chromium/chromium-wrapper"
+        if [ "$(dpkg-divert --listpackage "$conflicts")" = "rockchip-chromium-x11-utils" ]
```
```suggestion
        if [ "$(dpkg-divert --listpackage "$conflicts")" == "rockchip-chromium-x11-utils" ]
```
比较统一用==，避免=赋值的歧义

CodeChenL replied:
用==会lintian报错
```
W: rockchip-chromium-x11-utils: possible-bashism-in-maintainer-script '[ "$(dpkg-divert --listpackage "$conflicts")" ==' [preinst:9]
```

RadxaYuntian replied:
那行。我自己sh应该也是bash提供的，所以自己测试时没发现

RadxaYuntian commented on debian/rockchip-chromium-x11-utils.preinst:
```diff
@@ -0,0 +1,16 @@
+        ;;
+esac
+
+exit 0
```
```suggestion
exit 0

```

RadxaYuntian commented:
这个包也缺少codeowner

RadxaYuntian reviewed (changes requested):
rebase

RadxaYuntian commented on .github/CODEOWNERS:
codeowner放单独的commit里

CodeChenL committed:
fix: repair chromium installation failure

1. Breaks: rockchip-chromium-x11-utils (<< 0.2.0)
2. remove dpkg-divert from older package before install

CodeChenL committed:
feat: add CODEOWNERS
